cmd_test.go

```diff
@@ -0,0 +1,67 @@
+package orlop
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.ketch.com/lib/orlop/v2/config"
+	"go.uber.org/fx"
+)
+
+func TestRun(t *testing.T) {
+	os.Setenv("TEST_CONFIG_EMBEDDED", "true")
+	os.Setenv("TEST_CONFIG_REQ", "imhere")
+	os.Setenv("TEST_CONFIG_SLICED", "\"a\",\"b\",\"c\"")
+	os.Setenv("TEST_CONFIG_CUSTOM", "1m")
+	os.Setenv("TEST_CONFIG_MAP", "[\"a=b\",\"c=d\"]")
+	os.Setenv("TEST_CONFIG_HEX_ENCODED", "0102030405060708090A0B0C0D0E0F")
+	os.Setenv("TEST_CONFIG_BASE_64_ENCODED", "AQIDBAUGBwgJCgsMDQ4P")
+	os.Setenv("TEST_CONFIG_PTR", "123")
+
+	var cfg TestConfig
+
+	var module = fx.Options(
+		fx.Provide(
+			func(ctx context.Context, provider config.Provider) (*TestConfig, error) {
+				c, err := provider.Get(ctx, "config")
+				if err != nil {
+					return nil, err
+				}
+				return c.(*TestConfig), nil
```

Comment:
Should we standardise on using non-pointer config objects?

Reply:
May be not as we can't return a nil value in case of an error.

Reply:
In this case you could return `TestConfig{}, err`